### DISCUSSION TITLE
[@types/blazor__javascript-interop] missing methods

### DISCUSSION THREAD

**Discussion by Flachdachs** [DefinitelyTyped/DefinitelyTyped#65044] [Issues with a @types package] [answered]
The `DotNetObject` interface also has a [method `dispose`](https://github.com/dotnet/aspnetcore/blob/main/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L486) and [according to the documentation](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-7.0#component-instance-net-method-helper-class) it should be used.

Other missing functions in namespace `DotNet`:
* [createJSObjectReference](https://github.com/dotnet/aspnetcore/blob/main/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L135)
* [disposeJSObjectReference](https://github.com/dotnet/aspnetcore/blob/main/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L198)
* [createJSStreamReference](https://github.com/dotnet/aspnetcore/blob/main/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L158)

There are other exported things but only the functions above are [mentioned in the Blazor documentation](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-7.0#create-javascript-object-and-data-references-to-pass-to-net)

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
Hey @Flachdachs 
this particular package was an interim solution when Blazor didn't not ship with declaration files (if I recall correctly). So those types were created.
Today one can install '@microsoft/dotnet-js-interop' and this particular package ships declaration types amongst other things. The package is managed, published and maintained as part of Dotnet ecosystem AFAIK.
For example, the interface in subject, is now declared as:
```ts
export class DotNetObject {
        private _id;
        constructor(_id: number);
        invokeMethod<T>(methodIdentifier: string, ...args: any[]): T;
        invokeMethodAsync<T>(methodIdentifier: string, ...args: any[]): Promise<T>;
        dispose(): void;
        serializeAsArg(): {
            __dotNetObject: number;
        };
    }
```
thanks!

**Reply by Flachdachs** [reply]
Ah, ok, thanks for the reply. I didn't found the Microsoft package in my research. I suggest to remove the outdated package or give it a README.md to point to the official package. I will also reach out to the Microsoft documentation guys to mention TypeScript in the official Blazor documentation.
